src/hooks/wallets/useOnboard.ts

```diff
@@ -130,8 +131,19 @@ export const connectWallet = async (
   return wallets
 }
 
-export const switchWallet = (onboard: OnboardAPI) => {
-  connectWallet(onboard)
+export const switchWallet = async (onboard: OnboardAPI) => {
+  const oldWalletLabel = getConnectedWallet(onboard.state.get().wallets)?.label
+  const newWallets = await connectWallet(onboard)
+  const newWalletLabel = newWallets ? getConnectedWallet(newWallets)?.label : undefined
+
+  // If the wallet actually changed we disconnect the old connected wallet.
+  if (!newWalletLabel || oldWalletLabel !== ONBOARD_MPC_MODULE_LABEL) {
+    return
+  }
+
+  if (newWalletLabel !== oldWalletLabel) {
+    await onboard.disconnectWallet({ label: oldWalletLabel })
```

Comment:
So we dusconnect if the old wallet is social login? What if the new wallet is social login and the old one is MM? Will it work fine?

Reply:
Yes exactly. 
Only the Social signer will be disconnected if switched to another wallet.
We could also remove this behavior once we support every chain. But this is currently causing issues QA found.